src/classes/HiFiMixerSession.ts

```diff
@@ -852,86 +887,94 @@ export class HiFiMixerSession {
 
         return streamController.getAudioStream();
     }
+    /**
+     * Examines the underlying connection objects to determine
+     * whether or not we believe the connection to be, well,
+     * connected.
+     */
+    isConnected(): boolean {
+        return this._raviSession.getState() === RaviSessionStates.CONNECTED &&
+               this._raviSignalingConnection.getState() === RaviSignalingStates.OPEN;
+    }
+
+    /**
+     * Return the current state of the connection.
+     */
+    getCurrentHiFiConnectionState(): HiFiConnectionStates {
+        if (this._getUserFacingConnectionState) {
+            return this._getUserFacingConnectionState();
+        } else {
+            return undefined;
+        }
+    }
 
     /**
-     * Sets the state of the current connection, and
-     * fires the onChange handler if that state has, in fact, changed.
+     * Fire the onChange handler for a state operation
      * @param state
      */
-    _setCurrentHiFiConnectionState(state: HiFiConnectionStates): void {
-        if (this._currentHiFiConnectionState !== state) {
-            this._currentHiFiConnectionState = state;
+    async _onConnectionStateChange(state: HiFiConnectionStates, result: ConnectionAttemptResult): Promise<void> {
+        if (this.getCurrentHiFiConnectionState() !== state) {
             if (this.onConnectionStateChanged) {
-                this.onConnectionStateChanged(this._currentHiFiConnectionState);
+                this.onConnectionStateChanged(state, result);
             }
             this._hifiDiagnostics.fire();
```

Comment:
The intention with diagnostics is that we prime (both hifi and ravi diagnostics) on connect, and the fire hifiDiagnostics on anything else here.  That happened implicitly (without so much as a code comment, alas) before, with the state changes as they were.

Now that there are more states, one could guard this `fire()` here, but it would probably make the code clearer to _remove two `prime` lines (new lines 308, 309) in the audionet.init handler_,  and change this `fire()` line here to:
```
            if (state === HiFiConnectionStates.Connected) {
                this._raviDiagnostics.prime(this.mixerInfo.visit_id_hash);
                this._hifiDiagnostics.prime(this.mixerInfo.visit_id_hash);
            } else {
                this._hifiDiagnostics.fire();
            }
```

Reply:
@howard-stearns I made this change and will resolve this comment but please let me know if I didn't do it right!